jena-base/src/main/java/org/apache/jena/atlas/io/InStreamUTF8.java

```diff
@@ -144,7 +144,18 @@ public final int advance()
     public static final int advance(InputStreamBuffered input) {
         int x = input.advance() ;
         if ( x == -1 ) return -1 ;
-        return advance(input, x) ;
+
+        int codepoint = advance(input, x) ;
+        //System.out.printf("U+%04X\n", codepoint);
+
+        if ( false ) {
+//          if ( codepoint > Character.MAX_VALUE ) {
+//              throw new AtlasException("Out of range character (must use a surrogate pair)");
+//          }
```

Comment:
Delete? :point_up:

Reply:
OK - removed the commented text, rearrange and "if(false)" and "Character.isDefined" (that is a switch between being a pure UTF-8 decoder and one that only accepts characters in the Unicode version that Java supports).

(I'd forgotten that this class would get into the PR!)